### DISCUSSION TITLE
jQuery is not defined when trying to implement Gridstack.js

### DISCUSSION THREAD

**Discussion by fahadahmed** [remix-run/remix#4617] [Q&A]
### What version of Remix are you using?

1.7.0

### Steps to Reproduce

Trying to add an external js library reference in the head of the remix app via the following code:

```ts
export function canvasLinks() {
  return [
    { rel: 'stylesheet', href: styles },
    { rel: 'preload', href: "http://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js", as: "script", title: "jQuery"}
  ];
}
```

While the library is getting loaded in the head component for the remix app as seen in the screenshot below, the app crashes with the message `jQuery is not defined`.

<img width="305" alt="Screen Shot 2022-11-16 at 4 04 14 pm" src="https://user-images.githubusercontent.com/1190340/202088502-a2ab8248-fd67-47e9-af45-dfce970b33c0.png">

<img width="559" alt="Screen Shot 2022-11-16 at 3 46 58 pm" src="https://user-images.githubusercontent.com/1190340/202087820-03bca2fe-5fae-4e60-94d3-3199856c9991.png">

Want to implement a gridstack component which is dependant on jQuery library.

### Expected Behavior

The app should be able to pick up the uploaded library link from the head and should include jQuery in the codebase to allow gridstack.js to work.

### Actual Behavior

Getting the error message in the console: `jQuery is not defined`. The app then is unable to interact with the component.

**Comment by tlawrie** [comment]
👀 Im interested in understanding how this would work

**Comment by fahadahmed** [comment]
Are there any suggestions on how to make this happen?

**Comment by machour** [comment]
How are you calling gridstack.js? Can you show your route code?

**Comment by fahadahmed** [comment]
I have a component called `Canvas.clients.tsx` which includes the following code to draw the Gridstack component. 

```ts
import { GridStack, GridStackItem } from 'react-gridstack';
import styles from './Canvas.styles.css';

export function canvasLinks() {
  return [
    { rel: 'stylesheet', href: styles },
    { rel: 'preload', href: "http://ajax.googleapis.com/ajax/libs/jquery/1.7.1/jquery.min.js", as: "script", title: "jQuery"}
  ];
}

export default function Canvas() {
  return (
    <div>
      <GridStack cellHeight={50} verticalMargin={10}>
        <GridStackItem id="item_1" x={0} y={0} minHeight={2} minWidth={2}>
          First Item
        </GridStackItem>
        <GridStackItem id="item_2" x={0} y={2}>
          Second Item
        </GridStackItem>
      </GridStack>
    </div>
  )
}
```
The `Canvas` component is being called inside a manage route which is as below: 

```ts
export default function Index() {
  return (
    <div style={{ margin: '2rem' }}>
      This is where we manage the insight report - Create Charts and canvas
      here.
      <div
        style={{
          display: 'grid',
          gridTemplateColumns: '1fr 2fr',
          gridGap: '1rem',
        }}
      >
        <div>
          Widgets
          <Grid>
            <GraphTile
              data={data}
              title="Some Donut Insight"
              description={dummyDescription}
              graphType="donut"
            />
            <GraphTile
              data={timeseriesData}
              title="Some Timeseries Insight"
              description={dummyDescription}
              graphType="timeseries"
            />
          </Grid>
        </div>
        <div>
          Preview
          <ClientOnly fallback={<p>Loading...</p>}>
            {() => <Canvas />}
          </ClientOnly>
        </div>
      </div>
    </div>
  );
}
```
